tests/integration/sources/test_tar.py

```diff
@@ -1 +1,75 @@
-# TODO: add tar source integration test after executor lands
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import shutil
+import tarfile
+import textwrap
+from pathlib import Path
+
+import pytest
+import yaml
+
+import craft_parts
+from craft_parts import Action, Step
+
+_LOCAL_DIR = Path(__file__).parent
+
+
+def test_source_tar(new_dir):
+    _parts_yaml = textwrap.dedent(
+        """\
+        parts:
+          foo:
+            plugin: make
+            source: foobar.tar.gz
+        """
+    )
+
+    parts = yaml.safe_load(_parts_yaml)
+    src = _LOCAL_DIR / "data" / "foobar.tar.gz"
```

Comment:
Might be nicer to create a tar as part of the test (or as fixture) rather than track the blob

Reply:
Changed to create the tar file programmatically in the test.